app/controllers/admin/nooks_controller.rb

```diff
@@ -100,5 +104,15 @@ def check_availability
       end
       render :js, template: 'admin/nooks/availability'
     end
+
+    def remove_photo
+      remain_photos = requested_resource.photos # copy the array
```

Comment:
Photo deleting might be something that could be used in other places in the app, so moving photo deleting into the nook model probably makes sense here. It would also make it slightly easier to test.
In general, we prefer skinny controllers and fatter models. If your logic could conceivably be used elsewhere and it can be separated from the fact that it's behind a web server, it should probably be in the model.